File: connected-devices-python workspace/iot-device/apps/labs/module07/CoapServerConnector.py
'''
Created on Mar 7, 2019

@author: khars
'''

import getopt
import sys
from coapthon.server.coap import CoAP
from coapthon.server.coap import CoAP
from labs.module07.TempResourceHandler import TempResourceHandler
from coapthon.resources.resource import Resource
from labs.common.ConfigUtil import ConfigUtil
from labs.common import ConfigConst

client = None
'''

'''
class CoapServerConnector(CoAP):
    '''
    classdocs
    '''

    def __init__(self,host,port,config):
        '''
        Constructor
        '''
        CoAP.__init__(self, (host, port))
        self.add_resource("temperature/", TempResourceHandler(config = config)) #Add Temperature resource while initializing server
        
    
    def start(self):
        try:
            print("Starting Server...")
            self.listen(10)
        except KeyboardInterrupt:
            print("Server Shutdown")
            self.close()
            print("Exiting...")
        finally:
            self.close()
            
        

File: connected-devices-java workspace/iot-gateway/src/main/java/schooldomain/studentname/connecteddevices/labs/module07/CoapServerConnector.java
package schooldomain.studentname.connecteddevices.labs.module07;

import java.net.InetAddress;
import java.net.InetSocketAddress;
import java.net.SocketException;

import org.eclipse.californium.core.CoapServer;
import org.eclipse.californium.core.network.CoapEndpoint;
import org.eclipse.californium.core.network.EndpointManager;

import com.labbenchstudios.edu.connecteddevices.common.ConfigConst;
import com.labbenchstudios.edu.connecteddevices.common.ConfigUtil;

/**
 * This class creates an instance of TempResourceHandler and initializes the Server
 * 
 * @author khars
 *
 */
public class CoapServerConnector extends CoapServer {
	private final int COAP_PORT;
	
	public CoapServerConnector(ConfigUtil config) throws SocketException {
		COAP_PORT = config.getIntegerProperty(ConfigConst.COAP_GATEWAY_SECTION, ConfigConst.PORT_KEY);
		
		
		//Add the endpoints and the server resources while initializing the server
		addEndpoints();
		
		add(new TempResourceHandler());		

		
	}
	
	/**
	 * This method combines CoAP server to the end-point
	 */
	private void addEndpoints()
	{
		for (InetAddress addr : EndpointManager.getEndpointManager().getNetworkInterfaces()) {
    		// only binds to localhost
			if (addr.isLoopbackAddress()) {
				InetSocketAddress bindToAddress = new InetSocketAddress(addr, COAP_PORT);
				addEndpoint(new CoapEndpoint(bindToAddress));
			}
		}
	}
}